src/components/StatusBanner.tsx

```diff
@@ -1,31 +1,59 @@
 import React from "react";
-import { CSSTransition } from "react-transition-group";
 import "./StatusBanner.scss";
 
-type Props = {
-    mounted: boolean,
-    success: boolean,
-    message: string
+export type BootstrapAlertClass = 'alert-success' | 'alert-danger';
+
+export type BannerMessage = {
+    message: string;
+    type: BootstrapAlertClass;
+}
+
+type StatusBannerProps = {
+    bannerMessage?: BannerMessage;
+}
+
+type StatusBannerState = {
+    visible: boolean;
+    tempBannerMessage?: BannerMessage;
 }
 
-const SuccessBanner = (props: Props) => {
-    const hideClass = props.mounted ? "movedown" : null;
-    const bannerTypeClass = props.success ? "alert-success" : "alert-danger";
+const OPACITY_TRANSITION_DELAY_MS = 1000;
 
-    const getBannerMessage = (): string => {
-            if(props.success) {
-                return "Success!";
-            }
-            return props.message;
+export class StatusBanner extends React.Component<StatusBannerProps, StatusBannerState> {
+    state: StatusBannerState = {
+        visible: false
+    }
+
+    hideBanner = () => {
+        this.setState({
+            visible: false,
+            tempBannerMessage: this.props.bannerMessage
+        });
+        setTimeout(() => {
+            this.setState({
+                tempBannerMessage: undefined
+            });
+        }, 1000);
+    }
+
+    componentWillReceiveProps (nextProps: StatusBannerProps) {
+        if (this.state.visible && !nextProps.bannerMessage) {
+            this.hideBanner();
+        } else if (!this.state.visible && nextProps.bannerMessage) {
+            this.setState({
+                visible: true
+            });
         }
+    }
```

Comment:
This is a bit hacky but the status bar needs to maintain it's properties until it is fully hidden, so we let the component itself handle this change